### PR TITLE
ft  Sign up api verification email

### PR DESCRIPTION
**Pull request by hnobi**
# #160452475 Sign up api verification email 

#### What does this PR do?
Adds signup API controller, middleware and email verification function

#### Description of Task to be completed?
- Get sendGrid token
- Adds signup controller
- Adds signup route
- Adds signup validation middleware
- Adds test for signup controller and middleware
- Adds  function to generate one time token
- Adds a class for sending emails with sendGrid
#### How should this be manually tested?
- run `npm install` in the terminal  and run `localhost:3000/api/v1/auth/signup` on Postman
- check email inbox for verification link and copy the token e.g `nwmr7d3e7zgnwmr7d3e7zg` and then
-  run `localhost:3000/api/v1/auth/confirmation/nwmr7d3e7zgnwmr7d3e7zg`
#### Any background context you want to provide?
- When testing this feature, use Gmail  as sendGrid can't send mail to Yahoo mail
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/160452475
#### Screenshots (if appropriate)
Nil
#### Questions:
Nil

### CODE REVIEW

**Review by blackshady — COMMENTED** [review]
Nice job on the completion of this feature. Could you please delete `.vscode`

File: routes/auth/auth.js~HEAD [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Router } from 'express';
```
Comment (Alameen688) [inline routes/auth/auth.js~HEAD:1]: I think you have to remove this file. "auth.js~HEAD"

File: services/Mail.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      .send(msg)
+      .then((resp) => {
+        if (resp[0].statusCode === 202) {
+          return res.status(201).send({
+            status: 'success',
+            message: req.message,
+          });
+        }
+        return res.status(400).send({
+          status: 'failed',
```
Comment (Alameen688) [inline services/Mail.js:37]: Just so we would be consistent, use status: 'error'. @blackshady already used 'error' status on his login task and that is cool. I think we should just stick to that

**Comment by hnobi** [comment]
I closed this PR because it removed changes implemented by [@blackshady's pr](https://github.com/andela/odin-ah-backend/pull/10)

**Review by blackshady — COMMENTED** [review]
Nice job on this task.

**Pull request closed without merging**